.github/workflows/deploy-nightly-playground.yml

```diff
@@ -0,0 +1,105 @@
+
+          echo "OPENSEARCH_DIST_MANIFEST_URL=https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${DIST_VERSION}/$opensearch_build_id/linux/x64/tar/dist/opensearch/manifest.yml" >> "$GITHUB_OUTPUT"
+          echo "OPENSEARCH_DASHBOARDS_DIST_MANIFEST_URL=https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/${DIST_VERSION}/$dashboards_build_id/linux/x64/tar/dist/opensearch-dashboards/manifest.yml" >> "$GITHUB_OUTPUT"
+
+  validate-and-deploy:
+    outputs:
+      ENDPOINT: ${{ steps.get-endpoint.outputs.ENDPOINT }}
+      OPENSEARCH_JSON: ${{ steps.get-endpoint.outputs.OPENSEARCH_JSON }}
+      OPENSEARCH_DASHBOARDS_JSON: ${{ steps.get-endpoint.outputs.OPENSEARCH_DASHBOARDS_JSON }}
+    permissions:
+      id-token: write
+      contents: read
+    needs: set-os-osd-urls
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: 'opensearch-project/opensearch-build'
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.9
+
+      - name: Install Pipenv and Dependencies
+        run: |
+          python -m pip install --upgrade pipenv wheel
+          pipenv install --deploy --dev
+
+      - name: Validate OS and OSD
+        run: ./validation.sh --file-path opensearch=${{needs.set-os-osd-urls.outputs.OPENSEARCH_URL}} opensearch-dashboards=${{needs.set-os-osd-urls.outputs.OPENSEARCH_DASHBOARDS_URL}}
+
+      - uses: actions/checkout@v3
+
+      - uses: aws-actions/configure-aws-credentials@v4.0.2
+        with:
+          role-to-assume: ${{ secrets.DEPLOY_NIGHTLY_PLAYGROUND_ROLE }}
+          aws-region: us-west-2
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 16.x
+
+      - name: Deploy nightly playground
+        working-directory: nightly-playground
+        run: |
+          npm install
+          npm run cdk bootstrap -- -c distVersion=${DIST_VERSION} -c distributionUrl=${{needs.set-os-osd-urls.outputs.OPENSEARCH_URL}} -c dashboardsUrl=${{needs.set-os-osd-urls.outputs.OPENSEARCH_DASHBOARDS_URL}} -c dashboardPassword=${{ SECRETS.DASHBOARDS_PASSWORD }}
+          npm run cdk deploy "*" -- -c distVersion=${DIST_VERSION} -c distributionUrl=${{needs.set-os-osd-urls.outputs.OPENSEARCH_URL}} -c dashboardsUrl=${{needs.set-os-osd-urls.outputs.OPENSEARCH_DASHBOARDS_URL}} -c dashboardPassword=${{ SECRETS.DASHBOARDS_PASSWORD }} --require-approval never
+
+      - name: Get endpoint URL
+        id: get-endpoint
+        run: |
+          echo "ENDPOINT=$(aws cloudformation --region us-west-2 describe-stacks --stack-name infraStack-2x --query 'Stacks[0].Outputs[0].OutputValue' --output text)" >> "$GITHUB_OUTPUT"
```

Comment:
you can write cdk deploy output to a json file and read the endpoint from there as well.

Reply:
Done!